src/store/state.js

```diff
@@ -0,0 +1,67 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const state = createSlice({
+  name: 'state',
+  initialState: {
+    activityIndex: 0,
+    adIndex: 0,
```

Comment:
do you need really 3 indexes? 

can activity , ad and boardmessage don't use the same index?

Reply:
`scopeState`?

Reply:
I was thinking of `screenIndex`, i.e. the index of the screen that's currently visible

Reply:
ScreenIndex, sounds like index of the screen you are on, maybe currentIndex, of screenCurrentIndex.

The best choice here is consistency, so choose a name that is the same as the `current` prop in AdProps for example.

Maybe your statemachine component can also be renamed to `Screens`